### PR TITLE
SCANNPM-52 Fix scanner archive directory when having an architecture

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -45,6 +45,7 @@
         "pretty-quick": "4.0.0",
         "rimraf": "5.0.5",
         "sinon": "19.0.2",
+        "sonarqube-scanner": "^4.2.4",
```
Comment (7PH) [inline package-lock.json:48]: I don't think that's needed and it's removed when I run `npm i`

Comment (henryju) [reply package-lock.json:48]: Good catch, my mistake

File: src/scanner-cli.ts [hunk 2]
```diff
@@ -52,31 +52,33 @@ export function normalizePlatformName(): 'windows' | 'linux' | 'macosx' {
...
-  const version = properties[ScannerProperty.SonarScannerCliVersion] ?? SCANNER_CLI_VERSION;
-  if (!/^[\d.]+$/.test(version)) {
-    throw new Error(`Version "${version}" does not have a correct format."`);
+  const versionStr = properties[ScannerProperty.SonarScannerCliVersion] ?? SCANNER_CLI_VERSION;
+  const version = semver.coerce(versionStr);
+  if (!version) {
+    throw new Error(`Version "${versionStr}" does not have a correct format."`);
   }
+  const archStr = version.compare('6.1.0') >= 0 ? 'x64' : undefined;
+  let archSuffix = archStr ? `-${archStr}` : '';
```
Comment (7PH) [inline src/scanner-cli.ts:77]: NP: should be `const` but feel free to ignore

```suggestion
  const archSuffix = archStr ? `-${archStr}` : '';
```

**Review by 7PH — APPROVED** [review]
LGTM, validated with SQ 10.8-snapshot, 9.9 with default Scanner CLI and Scanner CLI 6.0.0.4432

**Pull request merged**